nx/lib/nx/pytorch_backend.ex

```diff
@@ -0,0 +1,177 @@
+defmodule Nx.PytorchBackend do
+  @behaviour Nx.Backend
+
+  defstruct [:ref]
+
+  alias Nx.Pytorch.NIF
+
+  import Nx.Shared
+
+  funs =
+    Nx.Backend.behaviour_info(:callbacks) --
+      [
+        tensor: 1,
+        iota: 2,
+        eye: 1,
+        random_uniform: 3,
+        random_normal: 3,
+        reshape: 3,
+        squeeze: 3,
+        add: 3,
+        dot: 5,
+        cholesky: 2,
+        from_binary: 3,
+        to_binary: 2,
+        backend_deallocate: 1,
+        inspect: 2
+      ]
+
+  for {fun, arity} <- funs do
+    args = Macro.generate_arguments(arity, __MODULE__)
+
+    def unquote(fun)(unquote_splicing(args)) do
+      raise "#{unquote(fun)}() is not supported"
+    end
+  end
+
+  ## Creation
+
+  @impl true
+  def tensor(tensor) do
+    tensor
+  end
+
+  def torch_type({:u, 8}), do: :byte
```

Comment:
Don't they support higher unsigned types? 🤔

Reply:
It's strange, but there are no definitions of higher unsigned types anywhere in the include files of the lib.
So, i.e. `kUInt32` is not available.

Here's what we've got in `types.h:

```
/// Fixed width dtypes.
constexpr auto kUInt8 = at::kByte;
constexpr auto kInt8 = at::kChar;
constexpr auto kInt16 = at::kShort;
constexpr auto kInt32 = at::kInt;
constexpr auto kInt64 = at::kLong;
constexpr auto kFloat16 = at::kHalf;
constexpr auto kFloat32 = at::kFloat;
constexpr auto kFloat64 = at::kDouble;
```

Reply:
Correct, PyTorch has 12 data types, and only 8bit int has unsigned version: https://pytorch.org/docs/stable/tensor_attributes.html

So, I'll map Nx unsigned ints to PyTorch's signed types for now. Should we issue any warning when unsigned PyTorch tensor is requested?

Reply:
I would raise for now, if a non-existing type is requested, then we can revisit it later. :)

Reply:
OK, changed to rising.